### PR TITLE
Add Normalcy hardfork

### PR DESCRIPTION
**Pull request by cffls**
Add Normalcy hardfork, which allows a chain to follow standard EVM in core execution, excluding pre/post-execution. Design rationale could be seen in this [doc](https://hackmd.io/4LplWzQkTQyQr0_SUrJCMQ).

### CODE REVIEW

**Review by revitteth — APPROVED** [review]
I can't access the doc but the changes look ok to me :)

**Comment by cffls** [comment]
> I can't access the doc but the changes look ok to me :)

Thanks for the quick response! I just opened the doc to public. You can try again now :D

File: erigon-lib/chain/chain_config.go [hunk 1]
```diff
@@ -124,7 +125,7 @@ type BorConfig interface {
 func (c *Config) String() string {
 	engine := c.getEngine()
 
-	return fmt.Sprintf("{ChainID: %v, Homestead: %v, DAO: %v, Tangerine Whistle: %v, Spurious Dragon: %v, Byzantium: %v, Constantinople: %v, Petersburg: %v, Istanbul: %v, Muir Glacier: %v, Berlin: %v, London: %v, Arrow Glacier: %v, Gray Glacier: %v, Terminal Total Difficulty: %v, Merge Netsplit: %v, Shanghai: %v, Cancun: %v, Prague: %v, Osaka: %v, Engine: %v, NoPruneContracts: %v}",
+	return fmt.Sprintf("{ChainID: %v, Homestead: %v, DAO: %v, Tangerine Whistle: %v, Spurious Dragon: %v, Byzantium: %v, Constantinople: %v, Petersburg: %v, Istanbul: %v, Muir Glacier: %v, Berlin: %v, London: %v, Arrow Glacier: %v, Gray Glacier: %v, Terminal Total Difficulty: %v, Merge Netsplit: %v, Shanghai: %v, Cancun: %v, Prague: %v, Osaka: %v, Normalcy: %v, Engine: %v, NoPruneContracts: %v}",
```
Comment (Stefan-Ethernal) [inline erigon-lib/chain/chain_config.go:128]: Optional nitpick: consider using `%t` to format all the boolean values in the string.

**Review by Stefan-Ethernal — APPROVED** [review]

**Review by mandrigin — CHANGES_REQUESTED** [review]
some questions about abstractions

**Review by mandrigin — APPROVED** [review]

**Comment by Sharonbc01** [comment]
Hey @mandrigin just checking as Jerry is out this week. Were you okay with his proposed changes and his write up on the Normalcy fork>  https://hackmd.io/@4cbvqzFdRBSWMHNeI8Wbwg/r1hKHp_S0

**Pull request merged**